programs/castle-vault/src/instructions/init.rs

```diff
@@ -179,53 +183,18 @@ impl<'info> Initialize<'info> {
     }
 }
 
-pub fn validate_fees(fees: &FeeArgs) -> ProgramResult {
-    if fees.fee_carry_bps > 10000 {
-        return Err(ErrorCode::FeeBpsError.into());
-    }
-
-    if fees.fee_mgmt_bps > 10000 {
-        return Err(ErrorCode::FeeBpsError.into());
-    }
-
-    if fees.referral_fee_pct > 50 {
-        return Err(ErrorCode::ReferralFeeError.into());
-    }
-
-    Ok(())
-}
-
-/// Creates a new vault
-///
-/// # Arguments
-///
-/// * `bumps` - bump seeds for creating PDAs
-/// * `strategy_type` - type of strategy that rebalance will execute
-/// * `fees` - carry and management fee that the vault collects denominated in basis points on behalf of primary and supplementary fee receivers
 pub fn handler(
     ctx: Context<Initialize>,
     bumps: InitBumpSeeds,
-    strategy_type: StrategyType,
-    rebalance_mode: RebalanceMode,
-    fees: FeeArgs,
-    vault_deposit_cap: u64,
-    allocation_cap_pct: u8,
+    config: VaultConfigArg,
 ) -> ProgramResult {
     let clock = Clock::get()?;
 
     // Validating referral token address
     ctx.accounts.validate_referral_token()?;
 
-    // Validating referral token account's mint
-    validate_fees(&fees)?;
-
-    // TODO compute the lower limit of the cap using number of lenging pools
-    if !(34..=100).contains(&allocation_cap_pct) {
-        return Err(ErrorCode::AllocationCapError.into());
-    }
-
     let vault = &mut ctx.accounts.vault;
-    vault.version = 1;
+    vault.version = 2;
```

Comment:
1. why do we increase the version here? I feel we can set the first stable version on main branch to 1
2. Is it too conservative to have 8 bits for the version?

Reply:
1. This change requires a major version bump since it is not backwards compatible, and the bump should be everywhere.
2. u8 is the smallest uint available